plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionFields.java

```diff
@@ -60,20 +66,32 @@ public static void parsePartitionField(PartitionSpec.Builder builder, String fie
     {
         @SuppressWarnings("PointlessBooleanExpression")
         boolean matched = false ||
-                tryMatch(field, IDENTITY_PATTERN, match -> builder.identity(match.group())) ||
-                tryMatch(field, YEAR_PATTERN, match -> builder.year(match.group(1))) ||
-                tryMatch(field, MONTH_PATTERN, match -> builder.month(match.group(1))) ||
-                tryMatch(field, DAY_PATTERN, match -> builder.day(match.group(1))) ||
-                tryMatch(field, HOUR_PATTERN, match -> builder.hour(match.group(1))) ||
-                tryMatch(field, BUCKET_PATTERN, match -> builder.bucket(match.group(1), parseInt(match.group(2)))) ||
-                tryMatch(field, TRUNCATE_PATTERN, match -> builder.truncate(match.group(1), parseInt(match.group(2)))) ||
-                tryMatch(field, VOID_PATTERN, match -> builder.alwaysNull(match.group(1))) ||
+                tryMatch(field, IDENTITY_PATTERN, match -> builder.identity(fromIdentifier(match.group(1)))) ||
+                tryMatch(field, YEAR_PATTERN, match -> builder.year(fromIdentifier(match.group(1)))) ||
+                tryMatch(field, MONTH_PATTERN, match -> builder.month(fromIdentifier(match.group(1)))) ||
+                tryMatch(field, DAY_PATTERN, match -> builder.day(fromIdentifier(match.group(1)))) ||
+                tryMatch(field, HOUR_PATTERN, match -> builder.hour(fromIdentifier(match.group(1)))) ||
+                tryMatch(field, BUCKET_PATTERN, match -> builder.bucket(fromIdentifier(match.group(1)), parseInt(match.group(2)))) ||
+                tryMatch(field, TRUNCATE_PATTERN, match -> builder.truncate(fromIdentifier(match.group(1)), parseInt(match.group(2)))) ||
+                tryMatch(field, VOID_PATTERN, match -> builder.alwaysNull(fromIdentifier(match.group(1)))) ||
                 false;
         if (!matched) {
             throw new IllegalArgumentException("Invalid partition field declaration: " + field);
         }
     }
 
+    // Currently, all Iceberg columns are stored in lowercase in the Iceberg metadata files.
+    // Unquoted identifiers support both lowercase as uppercase, and are converted to lowercase here.
+    // Quoted identifiers are restricted to lowercase only through the regex pattern, until we properly support quoted (case-sensitive) identifiers.
+    // See https://github.com/trinodb/trino/issues/12226#issuecomment-1128839259
```

Comment:
Link to https://github.com/trinodb/trino/issues/17 *and* to the new issue you create (https://github.com/trinodb/trino/pull/12227#pullrequestreview-989948944)